spec.html

```diff
@@ -28538,135 +28360,96 @@ <h1>Math.imul ( _x_, _y_ )</h1>
 
       <emu-clause id="sec-math.log">
         <h1>Math.log ( _x_ )</h1>
-        <p>Returns an implementation-approximated value representing the result of the natural logarithm of _x_.</p>
-        <ul>
-          <li>
-            If _x_ is *NaN*, the result is *NaN*.
-          </li>
-          <li>
-            If _x_ is less than 0, the result is *NaN*.
-          </li>
-          <li>
-            If _x_ is *+0* or *-0*, the result is *-&infin;*.
-          </li>
-          <li>
-            If _x_ is 1, the result is *+0*.
-          </li>
-          <li>
-            If _x_ is *+&infin;*, the result is *+&infin;*.
-          </li>
-        </ul>
+        <p>Returns the natural logarithm of _x_.</p>
+        <p>When the `Math.log` method is called with argument _x_, the following steps are taken:</p>
+        <emu-alg>
+          1. Let _n_ be ? ToNumber(_x_).
+          1. If _n_ is *NaN* or _n_ is *+&infin;*, return _n_.
+          1. If _n_ &lt; 0, return *NaN*.
+          1. If _n_ is *+0* or _n_ is *-0*, return *-&infin;*.
+          1. If _n_ is 1, return *+0*.
+          1. Return an implementation-approximated value representing the result of the natural logarithm of _n_.
+        </emu-alg>
       </emu-clause>
 
       <emu-clause id="sec-math.log1p">
         <h1>Math.log1p ( _x_ )</h1>
-        <p>Returns an implementation-approximated value representing the result of the natural logarithm of 1 + _x_. The result is computed in a way that is accurate even when the value of x is close to zero.</p>
-        <ul>
-          <li>
-            If _x_ is *NaN*, the result is *NaN*.
-          </li>
-          <li>
-            If _x_ is less than -1, the result is *NaN*.
-          </li>
-          <li>
-            If x is -1, the result is *-&infin;*.
-          </li>
-          <li>
-            If _x_ is *+0*, the result is *+0*.
-          </li>
-          <li>
-            If _x_ is *-0*, the result is *-0*.
-          </li>
-          <li>
-            If _x_ is *+&infin;*, the result is *+&infin;*.
-          </li>
-        </ul>
+        <p>Returns the natural logarithm of 1 + _x_. The result is computed in a way that is accurate even when the value of x is close to zero.</p>
+        <p>When the `Math.log1p` method is called with argument _x_, the following steps are taken:</p>
+        <emu-alg>
+          1. Let _n_ be ? ToNumber(_x_).
+          1. If _n_ is *NaN*, _n_ is *+0*, _n_ is *-0*, or _n_ is *+&infin;*, return _n_.
+          1. If _n_ &lt; -1, return *NaN*.
+          1. If _n_ is -1, return *-&infin;*.
```

Comment:
Better to go in descending order:
```suggestion
          1. If _n_ is -1, return *-&infin;*.
          1. If _n_ &lt; -1, return *NaN*.
```